src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -243,11 +244,13 @@ protected BlockDamageEvent simulateBlockDamagePure(Block block)
 		if (gamemode == GameMode.ADVENTURE || gamemode == GameMode.SPECTATOR)
 			return null;
 		Block block = location.getBlock();
+		BlockState blockState = block.getState();
+		block.setType(material);
 		BlockPlaceEvent event = new BlockPlaceEvent(block, null, null, null, this, true, null);
 		Bukkit.getPluginManager().callEvent(event);
-		if (!event.isCancelled())
+		if (event.isCancelled())
 		{
-			block.setType(material);
+			blockState.update(true);
```

Comment:
I don't think this will actually reset the material of the block. However, that might be a bug in MockBukkit:
[here we set the block state of a block](https://github.com/PeachesMLG/MockBukkit/blob/95aeaa4b7e32b156b789bcedf24eed9adb064fbe/src/main/java/be/seeseemelk/mockbukkit/block/BlockMock.java#L439), but [here we get the material of a block](https://github.com/PeachesMLG/MockBukkit/blob/95aeaa4b7e32b156b789bcedf24eed9adb064fbe/src/main/java/be/seeseemelk/mockbukkit/block/BlockMock.java#L156) (both methods are declared in the same class).

Reply:
Indeed an issue with MockBukkit fixed with #403.